components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml

```diff
@@ -384,33 +406,22 @@
           }).get();
         },
 
-        attachManagingListeners: function (baseUrl)
+        attachStateChangeCallbacks: function(callback)
         {
           var _this = this;
           var consentList = $(".consents");
           consentList.on("change", "li input", function (event)
           {
             var checkbox = $(event.target);
-            var rest = _this.interactWithManagingRest(checkbox.attr("data-bind"), baseUrl);
+            var name = checkbox.attr("data-bind");
             if (checkbox.is(":checked")) {
-              rest("grant");
+              callback(name, true);
             } else {
-              rest("revoke");
+              callback(name, false);
```

Comment:
You can add a comment like "pass name to callback". `if (x) true else false` is a common anti-pattern.

Another point I just noticed, you can eliminate jQuery from this function entirely, since all the features are implemented across all of our supported browsers.

```js
function (e) { callback(e.target.dataset.bind, !!e.target.checked) }
```